### PR TITLE
testing/firefox: enable on armhf, ppc64le and x86

### PR DESCRIPTION
**Pull request by J0WI**

### CODE REVIEW

**Comment by vicencb** [comment]
Hi J0WI,
on the aarch64 build log line 425:
```
mozbuild.configure.options.InvalidOptionError: --disable-elf-hack is not available in this configuration
```
Could you try removing `--disable-elf-hack` from APKBUILD?

Regards,
&nbsp;&nbsp;Vicente.

File: testing/firefox/APKBUILD [hunk 1]
```diff
@@ -100,7 +110,7 @@ build() {
 		--prefix=/usr \
 		\
 		--disable-crashreporter \
-		--disable-elf-hack \
+		$_disable-elf-hack \
```
Comment (vicencb) [inline testing/firefox/APKBUILD]: It took me some time to spot the difference: it's underscores instead of dashes.
`_disable_elf_hack`
`_disable-elf-hack`

**Comment by ncopa** [comment]
aarch64 fails with:
```
DEBUG: Creating `/tmp/conftestTQjtOh.rs` with content:
DEBUG: | pub extern fn hello() { println!("Hello world"); }
DEBUG: Executing: `/usr/bin/rustc --crate-type staticlib --target=aarch64-alpine-linux-musl -o /tmp/conftestNLxK6v.rlib /tmp/conftestTQjtOh.rs`
DEBUG: The command returned non-zero exit status 1.
DEBUG: Its error output was:
DEBUG: | error[E0463]: can't find crate for `std`
DEBUG: |   |
DEBUG: |   = note: the `aarch64-alpine-linux-musl` target may not be installed
DEBUG: | 
DEBUG: | error: aborting due to previous error
DEBUG: | 
DEBUG: | For more information about this error, try `rustc --explain E0463`.
ERROR: Cannot compile for aarch64-unknown-linux-musl with /usr/bin/rustc
The target may be unsupported, or you may not have
```

**Comment by vicencb** [comment]
If the failing command
```
/usr/bin/rustc --crate-type staticlib --target=aarch64-alpine-linux-musl ...
```
is changed to
```
/usr/bin/rustc --crate-type staticlib --target=aarch64-unknown-linux-musl ...
```
then it works fine.

From the comment on commit 6c8381ba967cd3842484965f3f7c1200a42a5539, it looks like the correct triplet should be `aarch64-unknown-linux-musl`.

So, removing `replace("-unknown-", "-alpine-").` in `fix-rust-target.patch` may improve things...

Regards,
&nbsp;&nbsp;Vicenç.

**Comment by J0WI** [comment]
`fix-rust-target.patch` is reuired for `x86_64`. I wonder if this should be defined here:
https://github.com/alpinelinux/aports/blob/1a43a53ec67e2c5ca5fa770026cd904d745f32a1/community/rust/APKBUILD#L89-L123

Maybe @Cogitri knows more about it.

**Comment by vicencb** [comment]
There is an upstream rust bug report regarding the triplets, so, until it is fixed a workaround is needed.
Can the fix-rust-target patch file be split in two: one the the arches requiring it and another one without the unknown -> alpine translation ?

**Comment by Cogitri** [comment]
Setting RUST_TARGET accordingly should fix it for now.

**Comment by Cogitri** [comment]
Ah, the new `fix-rust-target.patch` requires you to set the `RUST_TARGET` environment variable accordingly.

**Comment by J0WI** [comment]
Could you give this a try?

**Comment by Cogitri** [comment]
Could you maybe make a MR at gitlab? The CI isn't that useful here :/

**Comment by J0WI** [comment]
I limited this on `armv7` for now. It would be great to have this, #11996 and especially #11997 in the next release.

**Comment by J0WI** [comment]
:ping_pong: @Cogitri

**Comment by Cogitri** [comment]
6368b5fe45d62d338f16493d719a748d7ea117e1 - let's see how that goes on the builders. Thanks!

**Comment by Cogitri** [comment]
Argh, fails on the builders, I guess due to Rust having been bumped to 1.39.0 in the meantime. Here's the log: https://build.alpinelinux.org/buildlogs/build-edge-armhf/testing/firefox/firefox-70.0.1-r1.log

**Comment by J0WI** [comment]
I can try to upgrade to 71 which is released tomorrow.

**Comment by Cogitri** [comment]
Yup, hopefully that'll work. Thanks!

**Pull request closed without merging**